app/scienceapi/events/admin.py

```diff
@@ -1,5 +1,31 @@
 from django.contrib import admin
+from django.utils import timezone
+from django.utils.http import unquote
 
 from .models import Event
 
-admin.site.register(Event)
+
+@admin.register(Event)
+class EventAdmin(admin.ModelAdmin):
+
+    # Override add view so we can peek at the timezone they've entered and
+    # set the current time zone accordingly before the form is processed
+    def add_view(self, request, form_url='', extra_context=None):
+        if request.method == 'POST':
+            timezone.activate(request.POST.get('timezone'))
```

Comment:
This seems wrong. It will set the timezone for the current thread globally.

If you want to set the timezone offset on the two fields, do them in the pre save hooks for those form fields vs. globally like this.